pkg/cloudprovider/vsphere/instances.go

```diff
@@ -154,19 +156,25 @@ func (i *instances) InstanceExistsByProviderID(ctx context.Context, providerID s
 
 	// Check if node has been discovered already
 	uid := GetUUIDFromProviderID(providerID)
-	if _, ok := i.nodeManager.nodeUUIDMap[uid]; ok {
-		klog.V(2).Info("instances.InstanceExistsByProviderID() CACHED with ", uid)
-		return true, nil
-	}
-
 	err := i.nodeManager.DiscoverNode(uid, cm.FindVMByUUID)
 	if err == nil {
 		klog.V(2).Info("instances.InstanceExistsByProviderID() EXISTS with ", uid)
 		return true, nil
 	}
 
-	klog.V(4).Info("instances.InstanceExistsByProviderID() NOT FOUND with ", uid)
-	return false, err
+	if err != vclib.ErrNoVMFound {
+		klog.V(4).Info("instances.InstanceExistsByProviderID() NOT FOUND with ", uid, ". Err: ", err)
```

Comment:
the "NOT FOUND" should be removed from this log line

Reply:
yup, replaced with failed